src/navigation/gate.py

```diff
@@ -70,11 +70,62 @@ def spider_gate_trajectory(cls, approach_distance: float, gate: Gate, rover_posi
         closest_approach_point = possible_approach_points[approach_idx]
         victory_point = possible_approach_points[1 - approach_idx]
 
+        coordinates = GateTrajectory.gate_select_path(
+            rover_position, closest_prep_point, closest_approach_point, center, victory_point, gate
+        )
+
         # put the list of coordinates together
-        coordinates = np.array([closest_prep_point, closest_approach_point, victory_point])
-        coordinates = np.hstack((coordinates, np.zeros(coordinates.shape[0]).reshape(-1, 1)))
         return GateTrajectory(coordinates)
 
+    def gate_select_path(
+        rover_position: np.ndarray,
+        prep: np.ndarray,
+        approach: np.ndarray,
+        center: np.ndarray,
+        done: np.ndarray,
+        gate: Gate,
+    ):
+        """
+        Here is a reference for the points mentioned below https://github.com/umrover/mrover-ros/wiki/Navigation#searchtrajectory
+        :param rover_position: position vector of the rover,
+        :param prep: This is the closest prep point given from spider_gate_trajectory,
+        :param approach: This is the closest approach point given from spider_gate_trajectory ,
+        :param center: This is the mid point of the two gate posts,
+        :param done: This is the victory point given from spider_gate_trajectory,
+        :param gate: Gate object representing the gate that the rover will drive through,
+        :returns: This is a np.array which represents the coordinates of the selected path
+        """
+
+        # Get the shapes of both the posts
+        post_one_shape, post_two_shape = gate.pget_post_shape()
+
+        rover = rover_position[:2]
+
+        # try paths with successively more points until we have one that won't intersect
+        all_pts = np.vstack((prep, approach, center, done))
+        start_index = 2
```

Comment:
I think that this logic may be easier to follow if you instead call this variable something like 'pts_to_include' and then start it off at 2. Then instead of decrementing from it, increment it and then use [-pts_to_include : ]  (so the last pts_to_include points in the list get used).